src/boot/efi/console.c

```diff
@@ -87,15 +131,21 @@ EFI_STATUS console_key_read(uint64_t *key, uint64_t timeout_usec) {
                                 timer,
                                 TimerRelative,
                                 MIN(timeout_usec, watchdog_ping_usec) * 10);
-                if (err != EFI_SUCCESS)
+                if (err != EFI_SUCCESS) {
+                        FreePool(protocols);
+                        FreePool(events);
```

Comment:
please don't do manual clean-up, always do `_cleanup_free_` like the rest of the codebase

Reply:
`protocols` is a static variable used as a cache, I'm not sure it should be marked using `_cleanup_free_`. I think `_cleanup_free_` frees when the variable go out of scope but the scope of that static variable is still the end of the function.